File: server/Publications/Publications.Domain/Publications/Collection.cs
using Publications.Domain.Shared;
using Publications.Domain.Shared.Slugs;
using Publications.Domain.Shared.ValueObjects;
using Redis.OM.Modeling;

namespace Publications.Domain.Publications;

/// <summary>
/// Represents a logically grouped set of publications.
/// </summary>
[Document(IndexName = "collection-idx", StorageType = StorageType.Json, Prefixes = ["collection"])]
public class Collection: Entity<Collection>
{
    [Indexed] 
    public string Icon { get; set; } = string.Empty;

    [Searchable(Weight = 1.0)] 
    public string Name { get; set; } = null!;

    [Searchable(Weight = 0.8)] 
    public string Description { get; set; } = string.Empty;

    [Indexed(Sortable = true)] 
    public int PublicationsCount { get; set; }

    public override Collection UpdateSlug(IWordsService wordsService)
    {
        Slug = SlugService.Create(Name, Id.ToString(), IsoLanguageCode.English, wordsService);
        return this;
    }
}

File: client/src/types/common/collection.ts
import type { PublicationSummary } from '../publication-summary/publication-summary'
import type { PaginatedCollection } from './paginated-collection'

export interface ICollection {
  icon: string
  name: string
  description: string
  publicationsCount: number
  id: string
  slug: string
}

export interface IDetailedCollection {
  collection: ICollection
  publications: PaginatedCollection<PublicationSummary>
}
